sdk/storage/Azure.Storage.DataMovement.Files.Shares/tests/MockShareFileStorageResource.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Storage.Files.Shares;
+
+namespace Azure.Storage.DataMovement.Files.Shares.Tests
+{
+    internal class MockShareFileStorageResource : ShareFileStorageResource
```

Comment:
Is this really a mock? It seems more like it's just wrapping protected methods with `internal` so they can be more directly tested. I'm not sure this is the right naming to use.

Also, instead of making this functionality a subclass of the implementation for every resource type (especially when we go back to blobs testing), what if we used shared source to provide these internal wrappers once in-between `StorageResourceItem` and the specific implementation. E.g. the inheritance structure here is `StorageResource` > `StorageResourceItem` > `StorageResourceItemWithInternalAccess` > `ShareFileStorageResource` where StorageResourceItemWithInternalAccess is shared source compiled in and therefore we have internals access to it. This would let us write these wrappers once and get them for free when testing all resource items.

We'd want one for containers as well.

Reply:
Done. I could also do the containers right now in a different PR.